explorer/trace_testdata/full_trace.carbon

```diff
@@ -6,13 +6,65 @@
+// CHECK:STDOUT: return (n + 1);
+// CHECK:STDOUT: }
+// CHECK:STDOUT:
+// CHECK:STDOUT: }
 // CHECK:STDOUT: fn Main ()-> i32 {
 // CHECK:STDOUT: {
-// CHECK:STDOUT: return 0;
+// CHECK:STDOUT: return N.Foo(0);
 // CHECK:STDOUT: }
 // CHECK:STDOUT:
 // CHECK:STDOUT: }
 // CHECK:STDOUT: ********** resolving names **********
+
+
+package ExplorerTest api;
+
+// CHECK:STDOUT: --- declared `TestInterface` as `interface TestInterface` in `root` (full_trace.carbon:[[@LINE+1]])
+interface TestInterface {}
+
+// CHECK:STDOUT: --- declared `N` as `namespace N` in `root` (full_trace.carbon:[[@LINE+1]])
+namespace N;
+
+// CHECK:STDOUT: --- resolved `N` as `namespace N` in `root` (full_trace.carbon:[[@LINE+1]])
+fn N.Foo(n: i32) -> i32 {
+  return n + 1;
+// CHECK:STDOUT: --- declared `Foo` as `fn N.Foo` in `root` (full_trace.carbon:[[@LINE+1]])
+}
+
+fn Main() -> i32 {
+  return N.Foo(0);
+// CHECK:STDOUT: --- declared `Main` as `fn Main` in `root` (full_trace.carbon:[[@LINE+357]])
+// CHECK:STDOUT: ** resolving decl `interface TestInterface` (full_trace.carbon:[[@LINE-14]])
+// CHECK:STDOUT: --- marked `TestInterface` declared but not usable in `root`
+// CHECK:STDOUT: --- marked `TestInterface` usable in `root`
+// CHECK:STDOUT: --- declared `Self` as `Self` in `interface TestInterface` (full_trace.carbon:[[@LINE-17]])
+// CHECK:STDOUT: ** finished resolving decl `interface TestInterface` (full_trace.carbon:[[@LINE-18]])
+// CHECK:STDOUT: ** resolving decl `namespace N` (full_trace.carbon:[[@LINE-16]])
+// CHECK:STDOUT: --- marked `N` usable in `root`
+// CHECK:STDOUT: ** finished resolving decl `namespace N` (full_trace.carbon:[[@LINE-18]])
+// CHECK:STDOUT: ** resolving decl `fn N.Foo` (full_trace.carbon:[[@LINE-13]])
+// CHECK:STDOUT: --- resolved `N` as `namespace N` in `root` (full_trace.carbon:[[@LINE-17]])
+// CHECK:STDOUT: --- marked `Foo` declared but not usable in `root`
+// CHECK:STDOUT: --- declared `n` as `n` in `fn N.Foo` (full_trace.carbon:[[@LINE-19]])
+// CHECK:STDOUT: --- marked `Foo` usable in `root`
+// CHECK:STDOUT: ** resolving stmt `{return (n + 1);}` (full_trace.carbon:[[@LINE-18]])
+// CHECK:STDOUT: ** resolving stmt `return (n + 1);` (full_trace.carbon:[[@LINE-21]])
+// CHECK:STDOUT: ** finished resolving stmt `return (n + 1);` (full_trace.carbon:[[@LINE-22]])
+// CHECK:STDOUT: ** finished resolving stmt `{return (n + 1);}` (full_trace.carbon:[[@LINE-21]])
+// CHECK:STDOUT: ** finished resolving decl `fn N.Foo` (full_trace.carbon:[[@LINE-22]])
+// CHECK:STDOUT: ** resolving decl `fn Main` (full_trace.carbon:[[@LINE+338]])
+// CHECK:STDOUT: --- marked `Main` declared but not usable in `root`
+// CHECK:STDOUT: --- marked `Main` usable in `root`
+// CHECK:STDOUT: ** resolving stmt `{return N.Foo(0);}` (full_trace.carbon:[[@LINE+335]])
+// CHECK:STDOUT: ** resolving stmt `return N.Foo(0);` (full_trace.carbon:[[@LINE-24]])
+// CHECK:STDOUT: --- resolved `Foo` as `fn N.Foo` in `root` (full_trace.carbon:[[@LINE-25]])
```

Comment:
Similarly, I was expecting to see this here:
```
--- resolved `N` as `namespace N` in `root` (...)
```

Oh, I see what's wrong: `Resolve` calls `TryResolve`, which calls `TryResolveHere`, which doesn't do any tracing -- we never reach the trace line in `ResolveHere` when starting from `Resolve`. Perhaps you could move the trace output from `ResolveHere` to `TryResolveHere`.

Reply:
Thanks for pointing this out!